test/src/server/routes/collection.js

```diff
@@ -0,0 +1,221 @@
+		const newCollaborator = await createEditor();
+		const newData = {
+			collaborators: [newCollaborator.toJSON()],
+			description: 'new description',
+			entityType: 'Edition',
+			name: 'new collection name',
+			privacy: 'private'
+		};
+
+		const res = await agent.post(`/collection/${collectionJSON.id}/edit/handler`).send(newData);
+		const updatedCollection = await new UserCollection({id: collectionJSON.id}).fetch({withRelated: ['collaborators']});
+		const updatedCollectionJSON = updatedCollection.toJSON();
+
+		expect(res.status).to.equal(200);
+		expect(updatedCollectionJSON.name).to.equal(newData.name);
+		expect(updatedCollectionJSON.entityType).to.equal(newData.entityType);
+		expect(updatedCollectionJSON.description).to.equal(newData.description);
+		expect(updatedCollectionJSON.collaborators.length).to.equal(1);
+		expect(updatedCollectionJSON.collaborators[0].collaboratorId).to.equal(newCollaborator.get('id'));
+	});
+
+	it('should return status code 500 for incorrect entityType', async () => {
+		const data = {
+			collaborators: [collaborator.toJSON()],
+			description: 'some description',
+			entityType: 'incorrect',
+			name: 'collectionName',
+			privacy: 'public'
+		};
+		const res = await agent.post(`/collection/${collectionJSON.id}/edit/handler`).send(data);
+		expect(res.status).to.equal(500);
+	});
+
+	it('should return status code 500 for empty collection name', async () => {
+		const data = {
+			collaborators: [collaborator.toJSON()],
+			description: 'some description',
+			entityType: 'Author',
+			name: '',
+			privacy: 'public'
+		};
+		const res = await agent.post(`/collection/${collectionJSON.id}/edit/handler`).send(data);
+		expect(res.status).to.equal(500);
+	});
+
+	it('should return status 500 when trying to edit entityType of a non empty collection', async () => {
+		const author = await createAuthor();
+		await new UserCollectionItem({
+			bbid: author.get('bbid'),
+			collectionId: collectionJSON.id
+		}).save(null, {method: 'insert'});
+
+		const data = {
+			collaborators: [collaborator.toJSON()],
+			description: 'some description',
+			entityType: 'Edition',
+			name: 'collectionName',
+			privacy: 'public'
+		};
+		const res = await agent.post(`/collection/${collectionJSON.id}/edit/handler`).send(data);
+		expect(res.status).to.equal(500);
+	});
+
+	it('should throw error when unauthorized user tries to edit the collection', async () => {
+		await createEditor();
+		// The `agent` now has the sessionid cookie saved, and will send it
+		// back to the server in the next request:
+		const newAgent = await chai.request.agent(app);
+		await newAgent.get('/cb');
+		const newData = {
+			description: 'new description',
+			entityType: 'Edition',
+			name: 'new collection name',
+			privacy: 'private'
+		};
+
+		const res = await newAgent.post(`/collection/${collectionJSON.id}/edit/handler`).send(newData);
+		// TODO expect(res.status).to.equal(500);
```

Comment:
Same comments as above, better to skip this test with `xit(…`, and when you do get to implementing the check, also check the error message.